client/test/unit/specs/components/AppHeader.spec.js

```diff
@@ -70,4 +76,57 @@ describe('Unit | Component | AppHeader.vue', () => {
       expect(authenticationService.disconnect).to.have.been.called;
     });
   });
+
+  describe('#subscribe', () => {
+    beforeEach(() => {
+      sinon.stub(authenticationService, 'isAuthenticated');
+      sinon.stub(authenticationService, 'getAccessToken').returns('fake token');
+      sinon.stub(notificationService, 'success');
+      sinon.stub(subscriptionsApi, 'subscribe').resolves();
+    });
+
+    afterEach(() => {
+      authenticationService.isAuthenticated.restore();
+      authenticationService.getAccessToken.restore();
+      notificationService.success.restore();
+      subscriptionsApi.subscribe.restore();
+    });
+
+    it('should call the subscriptions API with good params when user is authenticated', () => {
+      // given
+      authenticationService.isAuthenticated.returns(true);
+
+      // when
+      component.subscribe();
+
+      // then
+      expect(subscriptionsApi.subscribe).to.have.been.calledWith('fake token');
+    });
+
+    it('should do nothing when user is not authenticated', () => {
+      // given
+      authenticationService.isAuthenticated.returns(false);
+
+      // when
+      component.subscribe();
+
+      // then
+      expect(subscriptionsApi.subscribe).to.not.have.been.called;
```

Comment:
D'après la [doc](http://chaijs.com/api/bdd/#not) (et la langue anglaise), c'est bien dans ce sens. Ou alors je n'ai pas compris la remarque.

Reply:
Fais une recherche sur IntelliJ de la chaîne ".not.", tu trouveras : 
- to.not.have.been
- to.have.not.been

Je suis allé vérifier sur chai.js (et tu as raison ;-)) :  
 /**
   * ### .not
   *
   * Negates all assertions that follow in the chain.
   *
   *     expect(function () {}).to.not.throw();
   *     expect({a: 1}).to.not.have.property('b');
   *     expect([1, 2]).to.be.an('array').that.does.not.include(3);
   *
   * Just because you can negate any assertion with `.not` doesn't mean you
   * should. With great power comes great responsibility. It's often best to
   * assert that the one expected output was produced, rather than asserting
   * that one of countless unexpected outputs wasn't produced. See individual
   * assertions for specific guidance.
   *
   *     expect(2).to.equal(2); // Recommended
   *     expect(2).to.not.equal(1); // Not recommended
   *
   * @name not
   * @namespace BDD
   * @api public
   */